src/main/kotlin/org/rust/lang/core/types/infer/TypeInference.kt

```diff
@@ -33,6 +35,7 @@ fun inferTypesIn(fn: RsFunction): RsInferenceResult =
 class RsInferenceResult(
     private val bindings: Map<RsPatBinding, Ty>,
     private val exprTypes: Map<RsExpr, Ty>,
+    private val resolvedPaths: Map<RsPathExpr, List<RsCompositeElement>>,
```

Comment:
Here we use `List<RsCompositeElement>` instead of list of bould elements because bound elements used only for type inference and also because is't hard get substitution for UFCS-like stuff. But it's not clean anyway...